lib/spack/llnl/util/filesystem.py

```diff
@@ -191,8 +191,9 @@ def getuid():
     if sys.platform == "win32":
         import ctypes
 
+        # If not admin, use the string name of the login as a unique ID
         if ctypes.windll.shell32.IsUserAnAdmin() == 0:
-            return 1
+            return os.getlogin()
```

Comment:
Is it okay that this isn't numeric?

Reply:
I don't think we do anything with it that requires it to be a number, although there might be confusion if there is the possibility of changing between `"1"` and `1`, for example.

Reply:
It's fine this isn't numeric, that said Peter makes a good point...
We need to revisit this and similar concepts in general to use Windows ACLS and UIDs so if this isn't causing errors I'd like to punt to that effort. That said @scheibelp I'll defer to your judgement here.

Reply:
I thought I saw a some places where we check `if fs.getuid() == 0` which I don't know if that is generally allowed comparing string vs int.

I have python 3.8 locally and it seems fine, even `"0" == 0` yields `False` which I think is correct. I don't see anywhere where we try to do `int(fs.getuid())` which I think is the more problematic case as you pointed out.

Reply:
This function still returns `0` (not `"0"`) in all cases where the user is an admin. I did `grep -i "getuid" -r lib/spack/spack`:

1. I mostly see this in tests testsing for `== 0` (so those are fine)
2. I see a check in `Stage` comparing this to `get_owner_uid` (so possibly this is an improvement?)
3. One use in `util.gpg` that calls `str()`  as part of making a storage path (so that is fine)

Overall except for [2] these are definitely fine (documentation on how we might return different types could be useful (or we could encode that as a mypy return type). We might also want to add documentation discussing whether/how the behavior of `getuid` and `get_owner_uid` match up.

Reply:
Nice, I am fine then.